### DISCUSSION TITLE
Disable "Start discussion" button until everything is filled

### DISCUSSION THREAD

**Discussion by EricKotato** [community/community#2883] [Discussions]
When opening a discussion, button "Start discussion" is always active, but it should be active only when title and body is filled, and category is selected.

Of course, if you'll try to start a discussion like that, you'll have an error:

![](https://user-images.githubusercontent.com/2903496/101589574-c46c3d80-39f9-11eb-876d-c775efebd047.png)

But I think it's better to enable button only once everything is filled, to avoid unnecessary page reloads.

**Comment by tomkerkhove** [comment]
Good input, adding @becca

**Comment by becca** [comment]
Are there page reloads/performance issues you're experiencing due to the current design implementation? How would disabling this button change the experience for starting a new discussion? Did you find this experience confusing?

**Reply by Griatch** [reply]
Not the original poster, but I pretty much _always_ forget to select the category before pressing `Start discussion`. Thinking about it, I think I subconciously assume that I'm writing in the category I currently had selected before creating the new post. 
I agree that a greyed out button would make more sense, but maybe the category button should also default to the latest category you filtered by?

**Reply by becca** [reply]
Perhaps! We don't want to create any burden for maintainers by allowing folks to open discussions in the wrong space (category) though especially when/if each category will have different structures (a question will look quite different than someone sharing a recent project win). There's definitely ways we could make this better so it helps to know your specific interaction here that needs optimization is around the new/extra `category` selection before starting a discussion ✨

**Reply by ash-development** [reply]
> Perhaps! We don't want to create any burden for maintainers by allowing folks to open discussions in the wrong space (category) though especially when/if each category will have different structures (a question will look quite different than someone sharing a recent project win). There's definitely ways we could make this better so it helps to know your specific interaction here that needs optimization is around the new/extra `category` selection before starting a discussion ✨

i definitely think that a seperate screen encouraging you to select a category would be better in the flow of creating a discussion thread

**Reply by EricKotato** [reply]
I think @Griatch said everything I wanted to say: I usually forget to select category, and as a result, I have unnecessary page reloading.

Also I have 2 suggestions:

1. Move category selector to the left of "Start discussion" button. That way it will be harder to forget that you haven't chosed a category.
2. Prefill category with selected category in "Discussions" tab, so it will look like this:
  2.1. Open Discussions tab in a repo.
  2.2. Select category you need.
  2.3. Press "New discussion" button.
After loading you should see that selected category is the same you selected when reading discussions. Of course, this applies only when you're selected category. To indicate that, button text could be possibly changed to "New discussion in `<category name>`" when category is selected.

**Reply by Griatch** [reply]
@RadRussianRus I like the idea of moving the category selector (while also defaulting it to your latest category-pick). If the concern (as @becca wrote) is accidental mis-posts, maybe it could help to relabel the `Start discussion` to `Start discussion in <category>` or somesuch as an extra reminder?

**Reply by joel-ekstep** [reply]
@becca It'd also be great if we could set a template for a post, so that the person starting a discussion has some gentle nudges along the way about what information is helpful. Such an ability exists for Github issues, if I understand correctly.
